src/python/pants/backend/native/config/environment.py

```diff
@@ -42,34 +42,54 @@ def resolve_platform_specific(self, platform_specific_funs):
     return fun_for_platform()
 
 
-class Executable(object):
+# NB: @abstractproperty requires inheriting from AbstractClass to work!
+class Executable(AbstractClass):
 
   @abstractproperty
   def path_entries(self):
     """A list of directory paths containing this executable, to be used in a subprocess's PATH.
 
     This may be multiple directories, e.g. if the main executable program invokes any subprocesses.
+
+    :rtype: list of str
+    """
+
+  @abstractproperty
+  def exe_filename(self):
+    """The "entry point" -- which file to invoke when PATH is set to `path_entries()`.
+
+    :rtype: str
     """
 
+  # TODO(#???): rename this to 'runtime_library_dirs'!
```

Comment:
Please fill in the issue link or else kill the unfinished #??? thought. A todo that itself is in a half-done state is just too much. Maybe fine when you're hacking locally, but you really should clean things up like this before posting for review. You may be making an effort at this though and just missed this one.

Reply:
If it was possible to run testing through the travis shards before pushing a commit, I would never have any such comments. I'll grant that magit in emacs makes it easy to stage hunks of files and not the whole diff, but since bootstrapping pants when running in a docker image takes a very long time, the only method of iteration I can see is to abuse travis runs on each commit. I feel that this has continually led to the assumption that I think `TODO(#???)` is an acceptable thing to have in code for review, which I don't, at all -- I left this particular comment as is because I was trying to test the new code against travis instead of context switching. This also applies to the `NB` I left for myself about `@abstractproperty`, which was mentioned above. I have been working on trying to fix the local testing situation but it is difficult to do all of these things at the same time.

Reply:
Removed the `(#???)`!